std/exception.d

```diff
@@ -2145,3 +2145,58 @@ pure nothrow @safe unittest
 
     auto infSlice = infinite[0 .. $]; // this would throw otherwise
 }
+
+
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
+ *
+ * Note that if you want to pass in special arguments to your exception, you
+ * obviously can't use this mixin and have to write the constructor manually.
+ */
+mixin template StandardExceptionConstructors()
```

Comment:
I'm way of calling it `stdException` since the module's name is also `std.exception` since the functionality is not really particularly related to anything else in std.exception. How about `basicExceptionCtors`?